src/universal/modules/outcomeCard/components/AddSoftTeamMember.js

```diff
@@ -0,0 +1,225 @@
+  if (Object.keys(errors).length) {
+    onError(errors.inviteeEmail);
+    return false;
+  }
+  return true;
+};
+
+class AddSoftTeamMember extends Component {
+  static propTypes = {
+    area: PropTypes.string.isRequired,
+    assignRef: PropTypes.instanceOf(Element),
+    atmosphere: PropTypes.object.isRequired,
+    dirty: PropTypes.bool.isRequired,
+    setDirty: PropTypes.func.isRequired,
+    closePortal: PropTypes.func.isRequired,
+    dispatch: PropTypes.func.isRequired,
+    isActive: PropTypes.bool.isRequired,
+    menuRef: PropTypes.instanceOf(Element),
+    projectId: PropTypes.string.isRequired,
+    error: PropTypes.any,
+    submitting: PropTypes.bool,
+    submitMutation: PropTypes.func.isRequired,
+    onCompleted: PropTypes.func.isRequired,
+    onError: PropTypes.func.isRequired,
+    styles: PropTypes.object.isRequired,
+    team: PropTypes.object.isRequired
+  };
+
+  state = {inviteeEmail: ''};
+  onSubmit = (e) => {
+    e.preventDefault();
+    const {inviteeEmail} = this.state;
+    const {
+      area,
+      assignRef,
+      atmosphere,
+      closePortal,
+      setDirty,
+      dispatch,
+      projectId,
+      submitting,
+      submitMutation,
+      onCompleted,
+      onError,
+      team
+    } = this.props;
+    if (submitting) return;
+
+    // validate
+    setDirty();
+    const {teamMembers, softTeamMembers, teamId} = team;
+    const assignees = teamMembers.concat(softTeamMembers);
+    const isValid = validateEmailAddress(inviteeEmail, assignees, onError);
+    if (!isValid) return;
+
+    // resolve
+    closePortal();
+    assignRef.focus();
+    const invitees = [{email: inviteeEmail}];
+    submitMutation();
+    const handleCompleted = (res) => {
+      onCompleted();
+      const {inviteTeamMembers: {newSoftTeamMembers, reactivatedTeamMembers}} = res;
+      const newSoftTeamMemberId = newSoftTeamMembers && newSoftTeamMembers[0].id;
+      const reactivatedTeamMemberId = reactivatedTeamMembers && reactivatedTeamMembers[0].id;
+      submitMutation();
+      const updatedProject = {
+        id: projectId,
+        assigneeId: newSoftTeamMemberId || reactivatedTeamMemberId
+      };
+      UpdateProjectMutation(atmosphere, updatedProject, area, onCompleted, onError);
+    };
+    InviteTeamMembersMutation(atmosphere, {invitees, teamId}, dispatch, onError, handleCompleted);
+  };
+
+  componentWillReceiveProps(nextProps) {
+    const {isActive, menuRef} = nextProps;
+    if (isActive !== this.props.isActive) {
+      if (isActive) {
+        this.inputRef.focus();
+      } else {
+        menuRef.focus();
+      }
+    }
+  }
+
+  onChange = (e) => {
+    const {dirty, error, onCompleted, onError, team} = this.props;
+    const inviteeEmail = e.target.value;
+    this.setState({inviteeEmail});
+    if (dirty) {
+      const assignees = team.teamMembers.concat(team.softTeamMembers);
+      const isValid = validateEmailAddress(inviteeEmail, assignees, onError);
```

Comment:
doing our own validation is actually cleaner than using redux-form. For single-field forms, i think rolling our own in the way to go.

Reply:
Yeah. It certainly is easier for little stuff like this. I think variance per-component is ok. It's fine to have options.